libcudacxx/test/public_headers/CMakeLists.txt

```diff
@@ -47,6 +47,10 @@ function(libcudacxx_add_public_header_test header)
   # Ensure that if this is an atomic header, we only include the right architectures
   string(REGEX MATCH "atomic|barrier|latch|semaphore|annotated_ptr|pipeline" match "${header}")
   if(match)
+    # Ensure that we only compile the header when we have some architectures enabled
+    if ("${architectures_at_least_sm70}" STREQUAL "")
```

Comment:
Should be able to us `if(NOT architectures_at_least_sm70)` since an empty string is equal to false in CMake